tests/unit/convert_hl7v2_fhir/controllers/er7/test_er7_extractor.py

```diff
@@ -0,0 +1,86 @@
+import pytest
+
+from convert_hl7v2_fhir.controllers.er7.er7_extractor import ER7Extractor
+from hl7apy.parser import parse_message
+
+from convert_hl7v2_fhir.controllers.er7.exceptions import (
+    InvalidNHSNumberError,
+    MissingNHSNumberError,
+    ER7ExtractorError,
+    MissingPatientClassError,
+    MissingAdmissionTypeError,
+    MissingTimeOfAdmissionError,
+    MissingFamilyNameError,
+    MissingGivenNameError,
+)
+
+RAW_ER7_MESSAGE_GOOD = "MSH|^~\&|SIMHOSP|SFAC|RAPP|RFAC|20200508130643||ADT^A01|5|T|2.3|||AL||44|ASCII\rEVN|A01|20200508130643|||C006^Wolf^Kathy^^^Dr^^^DRNBR^PRSNL^^^ORGDR|\rPID|1|2590157853^^^SIMULATOR MRN^MRN|2590157853^^^SIMULATOR MRN^MRN~2478684691^^^NHSNBR^NHSNMBR||Esterkin^AKI Scenario 6^^^Miss^^CURRENT||19890118000000|F|||170 Juice Place^^London^^RW21 6KC^GBR^HOME||020 5368 1665^HOME|||||||||R^Other - Chinese^^^||||||||\rPD1|||FAMILY PRACTICE^^12345|\rPV1|1|I|RenalWard^MainRoom^Bed 1^Simulated Hospital^^BED^MainBuilding^5|28b|||C006^Wolf^Kathy^^^Dr^^^DRNBR^PRSNL^^^ORGDR|||MED|||||||||6145914547062969032^^^^visitid||||||||||||||||||||||ARRIVED|||20200508130643||"
```

Comment:
First time I run this, I get a warning per message:

```
tests/unit/convert_hl7v2_fhir/controllers/er7/test_er7_extractor.py:17
  /Users/alex/src/il-hans-nems-subscriber/tests/unit/convert_hl7v2_fhir/controllers/er7/test_er7_extractor.py:17: DeprecationWarning: invalid escape sequence '\&'
    RAW_ER7_MESSAGE_GOOD = "MSH|^~\&|SIMHOSP|SFAC|RAPP|RFAC|20200508130643||ADT^A01|5|T|2.3|||AL||44|ASCII\rEVN|A01|20200508130643|||C006^Wolf^Kathy^^^Dr^^^DRNBR^PRSNL^^^ORGDR|\rPID|1|2590157853^^^SIMULATOR MRN^MRN|2590157853^^^SIMULATOR MRN^MRN~2478684691^^^NHSNBR^NHSNMBR||Esterkin^AKI Scenario 6^^^Miss^^CURRENT||19890118000000|F|||170 Juice Place^^London^^RW21 6KC^GBR^HOME||020 5368 1665^HOME|||||||||R^Other - Chinese^^^||||||||\rPD1|||FAMILY PRACTICE^^12345|\rPV1|1|I|RenalWard^MainRoom^Bed 1^Simulated Hospital^^BED^MainBuilding^5|28b|||C006^Wolf^Kathy^^^Dr^^^DRNBR^PRSNL^^^ORGDR|||MED|||||||||6145914547062969032^^^^visitid||||||||||||||||||||||ARRIVED|||20200508130643||"
```

It goes away on the second test execution, but comes back if I delete all the `__pycache__` directories, so it's obviously happening the first time the source is parsed.

It's barfing on this sequence: `"MSH|^~\&` because, I presume, an escaped `&` doesn't map to a valid unicode sequence.  Should that `\&` be a literal `\` followed by a literal `&`?  Or is there a binary sequence we need to escape properly here?